tools/crm_resource_runtime.c

```diff
@@ -1508,6 +1512,34 @@ cli_resource_execute(const char *rsc_id, const char *rsc_action, GHashTable *ove
 
     params = generate_resource_params(rsc, data_set);
 
+    /* add meta_timeout env needed by some resource agents */
+    child = first_named_child(rsc->xml, "operations");
+    for (child = first_named_child(child, XML_ATTR_OP);
```

Comment:
You can simply use "for (child = rsc->ops_xml;" here; ops_xml is the <operations> section, with any template or id-ref already expanded

Reply:
rsc->ops_xml contained the \<operations\> element, so I use "for (child = first_named_child(rsc->ops_xml, XML_ATTR_OP);".

FYI: "crm_log_xml_debug(rsc->ops_xml, "TDEBUG1: ");" output the following log.
```
   debug: pe_get_configured_timeout:    TDEBUG:    <operations>
   debug: pe_get_configured_timeout:    TDEBUG:      <op name="start" timeout="99s" interval="0" id="dummy3-start-0"/>
   debug: pe_get_configured_timeout:    TDEBUG:      <op name="monitor" interval="10s" timeout="88s" id="dummy3-monitor-10s"/>
   debug: pe_get_configured_timeout:    TDEBUG:      <op name="stop" interval="0" timeout="33s" id="dummy3-stop-0"/>
   debug: pe_get_configured_timeout:    TDEBUG:    </operations>
```